src/envoy/tcp/mixer/filter.cc

```diff
@@ -78,6 +78,11 @@ Network::FilterStatus Filter::onData(Buffer::Instance &data, bool) {
                  filter_callbacks_->connection(), data.length());
   received_bytes_ += data.length();
 
+  cached_dynamic_metadata_ = filter_callbacks_->connection()
```

Comment:
Can you explain why it is better to get this here, rather than the way it was done before?

Reply:
Envoy filters like mongo and mysql will clear the dynamic metadata everytime they start a new decoding cycle (from the onData() callback). They generate metadata for every query they see. But mixer sends reports periodically. So by the time it sends the report, the data in streamInfo.DynamicMetadata might get overwritten multiple times, to the point that we only have info about the last query and not every query since the last report. That said, this implementation above is wrong as it too is overwriting it..

Reply:
Addressed in 08fd80ef87e35e9ad52466d8ef2f7b97941d9eb1.